scaleway/instances.go

```diff
@@ -152,24 +156,58 @@ func (i *instances) GetZoneByNodeName(ctx context.Context, nodeName types.NodeNa
 // ===========================
 
 // instanceAddresses extracts NodeAdress from the server
-func instanceAddresses(server *scwinstance.Server) []v1.NodeAddress {
+func (i *instances) instanceAddresses(server *scwinstance.Server) []v1.NodeAddress {
 	addresses := []v1.NodeAddress{
 		{Type: v1.NodeHostName, Address: server.Hostname},
 	}
 
-	if server.PrivateIP != nil && *server.PrivateIP != "" {
+	if server.PublicIP != nil {
 		addresses = append(
 			addresses,
-			v1.NodeAddress{Type: v1.NodeInternalIP, Address: *server.PrivateIP},
-			v1.NodeAddress{Type: v1.NodeInternalDNS, Address: fmt.Sprintf("%s.priv.instances.scw.cloud", server.ID)},
+			v1.NodeAddress{Type: v1.NodeExternalIP, Address: server.PublicIP.Address.String()},
+			v1.NodeAddress{Type: v1.NodeExternalDNS, Address: fmt.Sprintf("%s.pub.instances.scw.cloud", server.ID)},
 		)
 	}
 
-	if server.PublicIP != nil {
+	var pnNIC *scwinstance.PrivateNIC
+	if i.pnID != "" {
+		for _, pNIC := range server.PrivateNics {
+			if pNIC.PrivateNetworkID == i.pnID {
+				pnNIC = pNIC
+				break
+			}
+		}
+	}
+
+	if pnNIC != nil {
+		region, _ := server.Zone.Region()
+		ips, err := i.ipam.ListIPs(&ListIPsRequest{
+			ProjectID:    &server.Project,
+			ResourceType: ResourceTypeInstancePrivateNIC,
+			ResourceID:   &pnNIC.ID,
+			Region:       region,
+		})
+		if err != nil {
+			klog.Errorf("unable to query ipam for node %s: %v", server.Name, err)
+			return addresses
+		}
+
+		if len(ips.IPs) > 0 {
+			addresses = append(
+				addresses,
+				v1.NodeAddress{Type: v1.NodeInternalIP, Address: ips.IPs[0].Address.IP.String()},
+			)
+		}
+		klog.Errorf("no private network ip for node %s", server.Name)
```

Comment:
Is there a missing `return addresses` or `else{}` above? otherwise this error will be shown even when there is a private network ip

Reply:
nice catch!